components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/batch_prediction_job_remote_runner.py

```diff
@@ -0,0 +1,140 @@
+     scenario so return failed
+     - If the batch prediction job is running, continue polling the status
+  Also retry on ConnectionError up to _CONNECTION_ERROR_RETRY_LIMIT times during
+  the poll.
+  """
+  client_options = {'api_endpoint': location + '-aiplatform.googleapis.com'}
+  client_info = gapic_v1.client_info.ClientInfo(
+      user_agent='google-cloud-pipeline-components',)
+
+  batch_prediction_job_uri_prefix = f"https://{client_options['api_endpoint']}/v1/"
+
+  # Initialize client that will be used to create and send requests.
+  job_client = aiplatform.gapic.JobServiceClient(
+      client_options=client_options, client_info=client_info)
+
+  # Instantiate GCPResources Proto
+  batch_prediction_job_resources = GcpResources()
+  batch_prediction_job_resource = batch_prediction_job_resources.resources.add()
+
+  # Check if the Custom job already exists
+  if path.exists(gcp_resources) and os.stat(gcp_resources).st_size != 0:
+    with open(gcp_resources) as f:
+      serialized_gcp_resources = f.read()
+      batch_prediction_job_resource = json_format.Parse(
+          serialized_gcp_resources, batch_prediction_job_resource)
+      batch_prediction_job_name = batch_prediction_job_resource.resource_uri[
+          len(batch_prediction_job_uri_prefix):]
+
+      logging.info(
+          'BatchPredictionJob name already exists: %s. Continue polling the status',
+          batch_prediction_job_name)
+  else:
+    parent = f'projects/{project}/locations/{location}'
+    job_spec = json.loads(payload, strict=False)
+    create_batch_prediction_job_response = job_client.create_batch_prediction_job(
+        parent=parent, batch_prediction_job=job_spec)
+    batch_prediction_job_name = create_batch_prediction_job_response.name
+
+    # Write the job proto to output
+    batch_prediction_job_resource.resource_type = 'BachPredictionJob'
+    batch_prediction_job_resource.resource_uri = f'{batch_prediction_job_uri_prefix}{batch_prediction_job_name}'
+
+    with open(gcp_resources, 'w') as f:
+      f.write(json_format.MessageToJson(batch_prediction_job_resource))
+
+  # Poll the job status
+  retry_count = 0
+  while True:
+    try:
+      get_batch_prediction_job_response = job_client.get_batch_prediction_job(
+          name=batch_prediction_job_name)
+      retry_count = 0
+    # Handle transient connection error.
+    except ConnectionError as err:
+      retry_count += 1
+      if retry_count < _CONNECTION_ERROR_RETRY_LIMIT:
+        logging.warning(
+            'ConnectionError (%s) encountered when polling job: %s. Trying to '
+            'recreate the API client.', err, batch_prediction_job_name)
+        # Recreate the Python API client.
+        job_client = aiplatform.gapic.JobServiceClient(
+            client_options=client_options)
+      else:
+        logging.error('Request failed after %s retries.',
+                      _CONNECTION_ERROR_RETRY_LIMIT)
+        # TODO(kevinbnaughton) propagate the error.
```

Comment:
you could assign this to ruifang who is going to help implementing this

Reply:
Done.